ui/ducks/send/send.js

```diff
@@ -1354,14 +1358,25 @@ const slice = createSlice({
             tokens,
             tokenAddressList,
             isProbablyAnAssetContract,
+            udTlds,
           } = action.payload;
-
+          let validUDDomain = false;
+          try {
+            udTlds.payload.forEach((tld) => {
+              if (state.recipientInput.toLowerCase().endsWith(`.${tld}`)) {
+                validUDDomain = true;
+              }
+            });
+          } catch {
+            validUDDomain = false;
+          }
           if (
             isBurnAddress(state.recipientInput) ||
             (!isValidHexAddress(state.recipientInput, {
               mixedCaseUseChecksum: true,
             }) &&
-              !isValidDomainName(state.recipientInput))
+              !isValidENSDomainName(state.recipientInput) &&
+              !validUDDomain)
```

Comment:
For legibility reasons could we add a new intermediate variable before this if block: `isValidDomain` which checks if it is either a valid ENS or a valid UNS domain, then we can simplify this check to isValidDomain === false